### PR TITLE
Fix incorrect tx link to tx explorer - fixes #5548

### LINKED ISSUE #5548: Incorrect tx link to tx explorer
### Description

In the dialog "Dispute details" and field "Delayed payout transaction sent to" it shows the donation address 34VLFgtFKAtwTdZ5rengTT2g2zC99sWQLC

However when clicking the link to see the transaction in an explorer it goes to an incorrect link. In my case using mempool.space it goes to https://mempool.space/tx/34VLFgtFKAtwTdZ5rengTT2g2zC99sWQLC

#### Version

v1.6.5

### Steps to reproduce

Go to Support->Arbitration and click in the Details icon for a given dispute. Then inside the dialog click the "Delayed payout transaction sent to" link

### Expected behaviour

The link should point to https://mempool.space/address/34VLFgtFKAtwTdZ5rengTT2g2zC99sWQLC
Even better it should point to the txid, not the donation address

### Actual behaviour

The link to the explorer is invalid

### Screenshots

image

#### Device or machine

Windows 10

### PR DESCRIPTION
**Pull request by ghost**
Fixes #5548

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/components/TxIdTextField.java [hunk 1]
```diff
@@ -173,7 +175,8 @@ private void openBlockExplorer(String txId) {
             BlockChainExplorer blockChainExplorer = isBsq ?
                     preferences.getBsqBlockChainExplorer() :
                     preferences.getBlockChainExplorer();
-            GUIUtil.openWebPage(blockChainExplorer.txUrl + txId, false);
+            String url = this.isAddress ? blockChainExplorer.addressUrl : blockChainExplorer.txUrl;
+            GUIUtil.openWebPage(url + txId, false);
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/TxIdTextField.java]: I don't think `TxIdTextField` should open an address URL. I think `AddressTextField` should do that just based on the name (I don't really know the code base yet)

File: desktop/src/main/java/bisq/desktop/main/overlays/windows/ContractWindow.java [hunk 2]
```diff
@@ -256,8 +258,9 @@ private void addContent() {
             addLabelTxIdTextField(gridPane, ++rowIndex, Res.get("shared.delayedPayoutTxId"), dispute.getDelayedPayoutTxId());
 
         if (dispute.getDonationAddressOfDelayedPayoutTx() != null) {
-            addLabelTxIdTextField(gridPane, ++rowIndex, Res.get("shared.delayedPayoutTxReceiverAddress"),
+            Tuple2<Label, TxIdTextField> field = addLabelTxIdTextField(gridPane, ++rowIndex, Res.get("shared.delayedPayoutTxReceiverAddress"),
                     dispute.getDonationAddressOfDelayedPayoutTx());
+            field.second.setAddress(true);
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/main/overlays/windows/ContractWindow.java]: I think `addAddressTextField()` should be used if you want to show an address

File: desktop/src/main/java/bisq/desktop/components/TxIdTextField.java [hunk 3]
```diff
@@ -66,6 +66,8 @@ public static void setWalletService(BtcWalletService walletService) {
     private TxConfidenceListener txConfidenceListener;
     @Setter
     private boolean isBsq;
+    @Setter
+    private boolean isAddress = false;
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/TxIdTextField.java]: A `TxId` cant be an address, this variable shouldn't be here

**Review by kaiwolfram — COMMENTED** [review]
NACK

`TxIdTextField` shouldnt be responsible for opening address URLs

File: desktop/src/main/java/bisq/desktop/components/TxIdTextField.java (hunk 1 above)
Comment (ghost) [reply desktop/src/main/java/bisq/desktop/components/TxIdTextField.java]: I agree - unfortunately  `AddressTextField` doesn't open web browser with blockchain explorer - it's purpose is to open wallet internally.

**Comment by ghost** [comment]
In accordance to comments by @KaiWitt I prepared alternative solution with new `ExplorerAddressTextField` instead of modifying `TxIdTextField`

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Constructor
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    public ExplorerAddressTextField() {
+        copyIcon = new Label();
+        copyIcon.setLayoutY(3);
+        copyIcon.getStyleClass().addAll("icon", "highlight");
+        copyIcon.setTooltip(new Tooltip(Res.get("txIdTextField.copyIcon.tooltip")));
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]:
```suggestion
        copyIcon.setTooltip(new Tooltip(Res.get("addressTextField.copyToClipboard")));
```

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        AwesomeDude.setIcon(copyIcon, AwesomeIcon.COPY);
+        AnchorPane.setRightAnchor(copyIcon, 30.0);
+
+        Tooltip tooltip = new Tooltip(Res.get("txIdTextField.blockExplorerIcon.tooltip"));
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: There is `tooltip.openBlockchainForAddress=Open external blockchain explorer for address: {0}` in displayStrings.properties, or maybe a new string is needed

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        blockExplorerIcon.setTooltip(tooltip);
+        AwesomeDude.setIcon(blockExplorerIcon, AwesomeIcon.EXTERNAL_LINK);
+        blockExplorerIcon.setMinWidth(20);
+        AnchorPane.setRightAnchor(blockExplorerIcon, 52.0);
+        AnchorPane.setTopAnchor(blockExplorerIcon, 4.0);
+
+        missingAddressWarningIcon = new Label();
+        missingAddressWarningIcon.getStyleClass().addAll("icon", "error-icon");
+        AwesomeDude.setIcon(missingAddressWarningIcon, AwesomeIcon.WARNING_SIGN);
+        missingAddressWarningIcon.setTooltip(new Tooltip(Res.get("txIdTextField.missingTx.warning.tooltip")));
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: needs a new dispay string, you're using a string for txIds

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        textField.setId("address-text-field");
+        textField.setEditable(false);
+        textField.setTooltip(tooltip);
+        AnchorPane.setRightAnchor(textField, 80.0);
+        AnchorPane.setLeftAnchor(textField, 0.0);
+        textField.focusTraversableProperty().set(focusTraversableProperty().get());
+        getChildren().addAll(textField, missingAddressWarningIcon, blockExplorerIcon, copyIcon);
+    }
+
+    public void setup(@Nullable String addressId) {
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]:
```suggestion
    public void setup(@Nullable String address) {
```

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+import lombok.Setter;
+
+import javax.annotation.Nullable;
+
+public class ExplorerAddressTextField extends AnchorPane {
+    private static Preferences preferences;
+
+    public static void setPreferences(Preferences preferences) {
+        ExplorerAddressTextField.preferences = preferences;
+    }
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: setters should be further down the file, after variable declarations, constructors and other methods

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        blockExplorerIcon.setOnMouseClicked(null);
+        copyIcon.setOnMouseClicked(null);
+        textField.setText("");
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Private
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    private void openBlockExplorer(String addressId) {
```
Comment (kaiwolfram) [inline desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]:
```suggestion
    private void openBlockExplorer(String address) {
```

File: desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java (hunk 8 above)
Comment (kaiwolfram) [reply desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: Or annotate `preferences` with `@Setter` and delete this method

Comment (ghost) [reply desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: `preferences` is static field, not sure why - I used same approach like in `TxIdTextField`

Comment (ripcurlx) [reply desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: > `preferences` is static field, not sure why - I used same approach like in `TxIdTextField`

Yes it is true that it is right now the same as in TxIdTextField, but I also would prefer using the @Setter annotation and getting rid of the unnecessary setPreferences method.

Comment (ghost) [reply desktop/src/main/java/bisq/desktop/components/ExplorerAddressTextField.java]: True - I thought it's not possible to use Setter on static field. Fixed it here and also in TxIdTextField

**Review by ripcurlx — APPROVED** [review]
ACK - Tested it on RegTest and it works as expected

**Pull request merged**